src/status_im/chat/subs.cljs

```diff
@@ -84,67 +84,25 @@
 
-(defn- intersperse-datemark
-  "Reduce step which expects the input list of messages to be sorted by clock value.
-  It makes best effort to group them by day.
-  We cannot sort them by :timestamp, as that represents the clock of the sender
-  and we have no guarantees on the order.
-
-  We naively and arbitrarly group them assuming that out-of-order timestamps
-  fall in the previous bucket.
-
-  A sends M1 to B with timestamp 2000-01-01T00:00:00
-  B replies M2 with timestamp    1999-12-31-23:59:59
-
-  M1 needs to be displayed before M2
-
-  so we bucket both in 1999-12-31"
-  [{:keys [acc last-timestamp last-datemark]} {:keys [timestamp datemark] :as msg}]
-  (cond (empty? acc)                                     ; initial element
-        {:last-timestamp timestamp
-         :last-datemark  datemark
-         :acc            (conj acc msg)}
-
-        (and (not= last-datemark datemark)               ; not the same day
-             (< timestamp last-timestamp))               ; not out-of-order
-        {:last-timestamp timestamp
-         :last-datemark  datemark
-         :acc            (conj acc {:value last-datemark ; intersperse datemark message
-                                    :type  :datemark}
-                               msg)}
-        :else
-        {:last-timestamp (max timestamp last-timestamp)  ; use last datemark
-         :last-datemark  last-datemark
-         :acc            (conj acc (assoc msg :datemark last-datemark))}))
-
-(defn sort-messages
-  "Remove hidden messages and sort by clock-value desc, breaking ties by message id"
-  [id->messages]
-  (->> id->messages
-       vals
-       (filter :show?)
-       (sort-by (juxt (comp unchecked-negate :clock-value) :message-id))))
-
-(defn- add-datemark [{:keys [timestamp] :as msg}]
-  (assoc msg :datemark (time/day-relative timestamp)))
-
-(defn- add-timestamp [{:keys [timestamp] :as msg}]
-  (assoc msg :timestamp-str (time/timestamp->time timestamp)))
-
-(defn intersperse-datemarks
-  "Add a datemark in between an ordered seq of messages when two datemarks are not
-  the same. Ignore messages with out-of-order timestamps"
-  [messages]
-  (when (seq messages)
-    (let [messages-with-datemarks (transduce (comp
-                                              (map add-datemark)
-                                              (map add-timestamp))
-                                             (completing intersperse-datemark :acc)
-                                             {:acc []}
-                                             messages)]
-      ; Append last datemark
-      (conj messages-with-datemarks {:value (:datemark (peek messages-with-datemarks))
-                                     :type  :datemark}))))
+(defn sort-message-groups
+  "Sorts message groups according to timestamp of first message in group"
+  [message-groups messages]
+  (sort-by
+   (comp unchecked-negate :timestamp (partial get messages) :message-id first second)
```

Comment:
This line needs some comments, especially `first second` part

Reply:
I will add comment